test/e2e/framework/workload.go

```diff
@@ -81,15 +81,15 @@ func WaitWorkloadPresentOnClusterFitWith(cluster, namespace, name string, fit fu
 	gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
 
 	klog.Infof("Waiting for Workload(%s/%s) synced on cluster(%s)", namespace, name, cluster)
-	gomega.Eventually(func() bool {
+	gomega.Eventually(func(g gomega.Gomega) (bool, error) {
 		workload, err := clusterClient.Resource(workloadGVR).Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
-			return false
+			return false, nil
 		}
 		typedObj := &workloadv1alpha1.Workload{}
 		err = runtime.DefaultUnstructuredConverter.FromUnstructured(workload.UnstructuredContent(), typedObj)
-		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-		return fit(typedObj)
+		g.Expect(err).ShouldNot(gomega.HaveOccurred())
+		return fit(typedObj), nil
 	}, pollTimeout, pollInterval).Should(gomega.Equal(true))
```

Comment:
The returned error is always nil, why add it?

Reply:
It's just a writing recommendation, and can refer to it here:
https://github.com/karmada-io/karmada/blob/ac2e9053561647fee7c854b2600f0c543c999c37/vendor/github.com/onsi/gomega/gomega_dsl.go#L277-L290

The all-assertion approach is also feasible and still needs some modifications.

Personally, I'm used to using the former.

Reply:
Thanks.